clients/index.js

```diff
@@ -300,6 +319,34 @@ function setupChannel(cb) {
     }
 
     self._controlServer.listen(self._controlPort, listenReady.signal);
+
+    function getHostForTChannel() {
+        var host = localIp();
+        var reason = validateHost(host);
+        if (reason !== null) {
+            nextAttempt(host, reason);
+            return;
+        }
+        self.tchannel.listen(self._port, host);
+    }
+
+    function nextAttempt(host, reason) {
+        var delay = self._getHostForTchannelAttemptBackoff * Math.pow(2, getHostAttemptCount);
```

Comment:
- the time limit is enforced by `_getHostForTchannelAttemptLimit`
- because of how the logic works, we never sleep for `delay * 2^10`, the last sleep is `delay *2^9`
- the total worst-case time slept (modulo fuzz) will then be `delay * 2^10 - 1` since ~17 minutes is maybe too much I'll lower the limit to 8 so that the upper bound drops to `delay * 2^8 - 1` or ~4:15